### PR TITLE
Add cachep2p@0.1.5 w/ npm auto-update

### LINKED ISSUE #9663: [Request] Add cachep2p
**Library name:** cachep2p
**Git repository url:** https://github.com/guerrerocarlos/CacheP2P
**License(s):** MIT
**Official homepage:** http://www.cachep2p.com/

### PR DESCRIPTION
**Pull request by extend1994**
Pull request for issue: #9663 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/guerrerocarlos/CacheP2P
 * Official website (optional, not the repository): https://www.cachep2p.com/
 * NPM package url (optional): https://www.npmjs.com/package/cachep2p
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 25
   - Count of stars: 761
   - Count of forks: 29
 * NPM download stats (optional):
   - Downloads in the last day: 1
   - Downloads in the last week: 15
   - Downloads in the last month: 64

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #9663, cc @guerrerocarlos

### CODE REVIEW

File: ajax/libs/cachep2p/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/guerrerocarlos/cachep2p.git"
+  },
+  "npmName": "cachep2p",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "cachep2p*.+(js)"
```
Comment (PeterDaveHello) [inline ajax/libs/cachep2p/package.json]: Let's use `"cachep2p*.+(js|css|map)"` for future compatibility and flexibility

**Comment by extend1994** [comment]
@PeterDaveHello I updated `package.json`, please help me review it, thank you.

File: ajax/libs/cachep2p/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/guerrerocarlos/cachep2p.git"
+  },
+  "npmName": "cachep2p",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "cachep2p*.+(js|css|map)"
```
Comment (PeterDaveHello) [inline ajax/libs/cachep2p/package.json]: By the way, you can also just use `cachep2p*` as there is no files with the prefix that we don't want to have.

File: ajax/libs/cachep2p/package.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  "name": "cachep2p",
+  "filename": "cachep2p.min.js",
+  "description": "The Peer to Peer Cache for the Masses",
+  "version": "0.1.5",
+  "author": {
+    "name": "Carlos Guerrero",
+    "email": "carlos@carlosguerrero.com",
+    "url": "https://cachep2p.com"
+  },
+  "homepage": "https://cachep2p.com",
```
Comment (PeterDaveHello) [inline ajax/libs/cachep2p/package.json]: This site looks down, seems `http://www.cachep2p.com/` is the working one, can you confirm that? Maybe we should create an issue at upstream repo if you have the same problem.

Comment (extend1994) [inline ajax/libs/cachep2p/package.json]: okay, I will confirm it by creating an issue!

Comment (PeterDaveHello) [inline ajax/libs/cachep2p/package.json]: You should confirm that you can't open it first?

File: ajax/libs/cachep2p/package.json (hunk 2 above)
Comment (extend1994) [inline ajax/libs/cachep2p/package.json]: okay, I get it :+1:

File: ajax/libs/cachep2p/package.json (hunk 3 above)
Comment (extend1994) [inline ajax/libs/cachep2p/package.json]: yes, I have confirmed that! And here is the issue about the site: guerrerocarlos/CacheP2P#24

Comment (extend1994) [inline ajax/libs/cachep2p/package.json]: @PeterDaveHello The site is current one and can be accessed now!

Comment (PeterDaveHello) [inline ajax/libs/cachep2p/package.json]: Great!

**Comment by PeterDaveHello** [comment]
@extend1994 so let's rebase with the latest master, then we can merge this.

File: ajax/libs/cachep2p/package.json (hunk 3 above)
Comment (guerrerocarlos) [inline ajax/libs/cachep2p/package.json]: Indeed, all good now. Thanks for doing all this.

**Comment by extend1994** [comment]
@PeterDaveHello I rebased it, please help me review this pull request, thank you.

**Pull request merged**